src/Uno.UWP/ApplicationModel/Chat/ChatMessageReader.Android.cs

```diff
@@ -0,0 +1,120 @@
+﻿#if __ANDROID__
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using Windows.Foundation;
+
+namespace Windows.ApplicationModel.Chat
+{
+	public partial class ChatMessageReader
+	{
+		// https://stackoverflow.com/questions/848728/how-can-i-read-sms-messages-from-the-device-programmatically-in-android
+
+		private TimeSpan _range;
+		private Android.Database.ICursor _cursor = null;
+		private int _colBody, _colTime, _colInOut, _colRead, _colFrom, _colTo, _colSeen, _colAddr, _colStatus;
+
+		internal ChatMessageReader(TimeSpan range)
+		{
+			_range = range;
+			Android.Content.ContentResolver cr = Android.App.Application.Context.ContentResolver;
+			//Android.Net.Uri messageUri = Android.Net.Uri.Parse("content://sms/");
+
+			_cursor = cr.Query(Android.Provider.Telephony.Sms.ContentUri, null, null, null, Android.Provider.Telephony.Sms.DefaultSortOrder);
+
+			if (_cursor.MoveToFirst())
+			{ // runtime optimizations
+				_colBody = _cursor.GetColumnIndex(Android.Provider.Telephony.TextBasedSmsColumns.Body);
+				_colTime = _cursor.GetColumnIndex(Android.Provider.Telephony.TextBasedSmsColumns.Date); //int
+				_colInOut = _cursor.GetColumnIndex(Android.Provider.Telephony.TextBasedSmsColumns.Type); // 1:inbox, 4:outbox, 2:sent, ...

```

Comment:
What are the purpose of the comments ?

Reply:
This comments helped me while implementing (info extracted from Android documentation), it could be useful if someone wants to deal with this code.